src/nvim/grid.c

```diff
@@ -1085,16 +1085,35 @@ void grid_del_lines(ScreenGrid *grid, int row, int line_count, int end, int col,
 }
 
 static void grid_draw_bordertext(VirtText vt, int col, int winbl, const int *hl_attr,
-                                 BorderTextType bt)
+                                 BorderTextType bt, int over_flow)
 {
+  int default_attr = hl_attr[bt == kBorderTextTitle ? HLF_BTITLE : HLF_BFOOTER];
+  if (over_flow > 0) {
+    grid_line_puts(1, "<", -1,  hl_apply_winblend(winbl, default_attr));
+    col += 1;
+  }
+
   for (size_t i = 0; i < kv_size(vt);) {
     int attr = -1;
     char *text = next_virt_text_chunk(vt, &i, &attr);
     if (text == NULL) {
       break;
     }
     if (attr == -1) {  // No highlight specified.
-      attr = hl_attr[bt == kBorderTextTitle ? HLF_BTITLE : HLF_BFOOTER];
+      attr = default_attr;
+    }
+    if (over_flow > 0) {
```

Comment:
this code will be hard to understand in the future. now is the time to add a 1-line comment that says what it does. As always, a brief input/output example is 100x more helpful than a long comment:

    // Example: "..." => "..."

is there not a truncate function that can be used? `trunc_string` looks promising: https://github.com/neovim/neovim/blob/64cf63a88110459d70d140f6074fa014adaef331/src/nvim/message.c#L518